src/environ/_environ_config.py

```diff
@@ -26,6 +26,26 @@
 log = logging.getLogger(CNF_KEY)
 
 
+# We define a sentinel for when prefixes are not set and special handling for
+# what value to use at the app level when that is the case
+class Sentinel(object):
+    def __init__(self, bool_=True):
+        self._bool = bool(bool_)
+
+    def __bool__(self):
+        return self._bool
+
+    __nonzero__ = __bool__  # For Python 2 compatibility
+
+
+PREFIX_NOT_SET = Sentinel(False)
+DEFAULT_PREFIX = "APP"
+
+
+def _get_prefix(obj):
```

Comment:
Please add a docstring that explains what the expression does. Just informal, so that the reader doesn't have to start evaluating it in their heads. :)

Reply:
Done, I'll let you resolve